sftp/1.3/modules/ROOT/edits/sftp-connection-dynamically.adoc

```diff
@@ -0,0 +1,147 @@
+. On the *General* tab, set the following parameters:
++
+* *Host*: `All Interfaces [0.0.0.0] (default)`
+* *Port*: `8081`
++
+. Click *OK*.
+. Drag a *Set Variable* component to the right of the *HTTP Listener* source.
+. Set the *Name* field to `content`, and the *Value* field to `#[payload]`.
+. Drag the File Connector *Read* operation to the right of the *Set Variable* component.
+. Set the *File Path* field to `recipients.csv`. +
+This reads a CSV file that contains a random set of SFTP destinations with columns such as host, user, and port.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of File Connector in the app.
+. Accept the default settings and click *OK*.
+. On the *MIME Type* tab of the *Read* operation, for the *MIME Type* field select `application/csv`.
+
+=== Configure Additional Components and Operations
+
+Continue creating your Mule application by adding additional components and operations:
+
+. In Studio, drag a *Transform Message* component to the right of the *Read* operation.
+. In the *Output* section of the component, add the following DataWeave code to map the columns of the CSV file:
++
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/java
+---
+payload map using (row = $) {
+   host: row.Host,
+   user: row.User,
+   password: row.Password}
+----
++
+. Drag a *For Each* component to the right of the *Transform Message* component.
+. Drag an *SFTP Write* operation into the *For Each* component. +
+On each For Each iteration, the *Write* operation writes the contents into each of the SFTP destinations.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of SFTP Write operation in the app, and set the following parameters:
++
+* *Host*: `#[payload.host]`
+* *Username*: `#[payload.username]`
+* *Password*: `#[payload.password]`
++
+[start=6]
+. Click *OK*.
+. On the configuration screen, set the *Path* field to `demo.txt`.
+. Set the *Content* field to `payload`.
+. Drag a *Set Payload* component to the right of the *For Each* component.
+. Set the *Value* field to `Multicast Ok`.
+
+=== Run and Test your Mule Application
+
+After creating your Mule application, run and test it:
+
+. In Studio, save your Mule app.
+. Click the project name in *Package Explorer* and then click *Run* > *Run As* > *Mule Application*. +.
+. Open a browser and type `+http://0.0.0.0:8081/multitenant+`. +
+
+Note on Mule 4 behavior for Mule 3 users:
+
+* The information posted through the HTTP Listener source is written to each SFTP site multiple times. Because the listener makes use of the xref:mule-runtime::streaming-about.adoc[repeatable streams feature], you do not need to worry about consuming the stream multiple times.
```

Comment:
linebreak
```suggestion
* The information posted through the HTTP Listener source is written to each SFTP site multiple times. 
+
Because the listener makes use of the xref:mule-runtime::streaming-about.adoc[repeatable streams feature], you do not need to worry about consuming the stream multiple times.
```